### PR TITLE
Updated the UIs of the assign groups and assign users in the edit roles interface

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Updated the UIs of the assign groups and assign users in the edit roles interface.

### Final UI
Screenshot 2023-10-13 at 13 26 48

### Other Notices
- These newly updated UIs will not be enabled from this PR.
- `role-permission.tsx` will be removed from a future PR.
- The `Autocomplete` component will come to oxygen (https://github.com/wso2/oxygen-ui/pull/165), until this PR merged will use the `Autocomplete` component in mui. Will update the component in a future a PR.

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+                                            )) }
+                                            renderOption={ (
+                                                props: HTMLAttributes<HTMLLIElement>, 
+                                                option: GroupsInterface
+                                            ) => (
+                                                <AutoCompleteRenderOption
+                                                    displayName={ RoleManagementUtils.getGroupDisplayName(option) }
+                                                    userstore={ RoleManagementUtils.getUserStore(option.displayName) }
+                                                    renderOptionProps={ props }
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:384]:
```suggestion
```

Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/94d3b881019fe6aaebe002549eb539dceec61d3d

File: apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import { ChipProps } from "@oxygen-ui/react";
+import Chip from "@oxygen-ui/react/Chip";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { GroupsInterface } from "apps/console/src/features/groups/models";
+import { UserBasicInterface } from "apps/console/src/features/users/models";
+import React, { FunctionComponent, ReactElement, SyntheticEvent, useState } from "react";
+import { ChipMoreDetails } from "./chip-more-details";
+
+interface RenderChipInterface extends IdentifiableComponentInterface, ChipProps {
+
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/94d3b881019fe6aaebe002549eb539dceec61d3d

File: apps/console/src/features/roles/components/edit-role/edit-role-common/chip-more-details.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                                        icon={ getSidePanelIcons().userStore }
+                                        verticalAlign="middle"
+                                    />
+                                ) }
+                                label={ userStore }
+                            />
+                        </Grid>
+                    </Grid>
+                </CardContent>
+
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-common/chip-more-details.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/94d3b881019fe6aaebe002549eb539dceec61d3d

File: apps/console/src/features/groups/api/groups.ts [hunk 4]
```diff
@@ -36,18 +36,17 @@ import {
...
-    .httpRequest.bind(AsgardeoSPAClient.getInstance())
-    .bind(AsgardeoSPAClient.getInstance());
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance());
 
 /**
  * Retrieve the list of groups in the system.
  *
- * @param domain user store
+ * @param domain - user store
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/groups/api/groups.ts:45]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/94d3b881019fe6aaebe002549eb539dceec61d3d

File: apps/console/src/features/roles/components/edit-role/edit-role-common/chip-more-details.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                            <Grid>
+                                <AppAvatar
+                                    image={ (
+                                        <AnimatedAvatar
+                                            name={ primaryText }
+                                            size="mini"
+                                        />
+                                    ) }
+                                    size="mini"
+
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-common/chip-more-details.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/48729e0b4d4f83c86a42653fade4b2b17144e384

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+    const {
+        data: originalGroupList,
+        isLoading: isGroupListFetchRequestLoading,
+        error: groupListFetchRequestError,
+        mutate: mutateGroupListFetchRequest
+    } = useGroupList(null, null, groupSearchValue ? `displayName co ${ groupSearchValue }` : null);
+
+    /**
+     * set initial selected groups options
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:90]: Check the other places 

```suggestion
     * Set initial selected groups options
```

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 7]
```diff
@@ -0,0 +1,547 @@
...
+    /**
+     * Handle the restore groups.
+     * 
+     * @param remainingGroups - remaining groups
+     */
+    const handleRestoreGroups = (remainingGroups: GroupsInterface[]) => {
+        const removedGroups: GroupsInterface[] = [];
+
+        removedGroupsOptions.forEach((group: GroupsInterface) => {
+            if (!remainingGroups.find((newGroup: GroupsInterface) => newGroup.id === group.id)) {
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:180]: Shall we add nullable checks?

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 8]
```diff
@@ -0,0 +1,547 @@
...
+                            )
+                            : null
+                    }
+                />
+            );
+        }
+    };
+
+    /**
+     * The following function handles the search query for the groups list.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:234]: Let's make the comments consistent
```suggestion
     *Handle the search query for the groups list.
```

**Review by dasuni-30 — APPROVED** [review]

**Comment by JayaShakthi97** [comment]
Bump the identity-apps-core and myaccount as well, since the theme, i18n modules are used in them.

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 9]
```diff
@@ -0,0 +1,547 @@
...
+            removeOperations.push({
+                "op": "remove",
+                "path": `groups[value eq ${ group.id }]`
+            });
+        } );
+
+        operation.data.Operations.push(...removeOperations);
+
+        selectedGroupsOptions?.map((group: GroupsInterface) => {
+            if(!initialSelectedGroupsOptions?.find((selectedGroup: GroupsInterface) => selectedGroup.id === group.id)) {
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:274]:
```suggestion
            if (!initialSelectedGroupsOptions?.find((selectedGroup: GroupsInterface) => selectedGroup.id === group.id)) {
```

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 10]
```diff
@@ -0,0 +1,547 @@
...
+                                            ) => (
+                                                <AutoCompleteRenderOption
+                                                    displayName={ RoleManagementUtils.getGroupDisplayName(option) }
+                                                    userstore={ RoleManagementUtils.getUserStore(option.displayName) }
+                                                    renderOptionProps={ props }           
+                                                />
+                                            ) }
+                                        />
+                                    )
+                                    : (
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:388]:
```suggestion
                                    ) : (
```

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 11]
```diff
@@ -0,0 +1,547 @@
...
+                                                />
+                                            )) }
+                                            renderOption={ (
+                                                props: HTMLAttributes<HTMLLIElement>, 
+                                                option: GroupsInterface
+                                            ) => (
+                                                <AutoCompleteRenderOption
+                                                    displayName={ RoleManagementUtils.getGroupDisplayName(option) }
+                                                    userstore={ RoleManagementUtils.getUserStore(option.displayName) }
+                                                    renderOptionProps={ props }
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:383]:
```suggestion
                                                    renderOptionProps={ props } 
```

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx [hunk 12]
```diff
@@ -0,0 +1,547 @@
...
+                                                        initialSelectedGroupsOptions?.find(
+                                                            (group: GroupsInterface) => group.id === option.id
+                                                        )
+                                                            ? "solid"
+                                                            : "outlined"
+                                                    }
+                                                />
+                                            )) }
+                                            renderOption={ (
+                                                props: HTMLAttributes<HTMLLIElement>,
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx:441]:
```suggestion
                                                props: HTMLAttributes<HTMLLIElement>,
```

Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 11 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 10 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 9 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 8 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 7 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

File: apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx (hunk 6 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa

**Review by dasuni-30 — APPROVED** [review]

File: apps/console/src/features/roles/components/wizard/role-permission.tsx [hunk 13]
```diff
@@ -236,7 +235,8 @@ export const PermissionList: FunctionComponent<PermissionListProp> = (props: Per
      * the tree nodes.
      * @param eventObject - event object
      */
-    const switcherIcon = eventObject => {
+    // eslint-disable-next-line @typescript-eslint/typedef
+    const switcherIcon = (eventObject) => {
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard/role-permission.tsx:239]: We can't. Either way this file will be removed from a future PR, that's why I used eslint disable.

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**